### PR TITLE
implement & test delete_channel

### PR DESCRIPTION
**Pull request by FranckNdame**
Implemented and tested the `delete_channel` feature.
As this is a quite different case with the absence of states I used `list_channels` as a helper function.

### CODE REVIEW

File: moto/backends.py [hunk 1]
```diff
@@ -80,6 +80,8 @@
     "support": ("support", "support_backends"),
     "mediaconnect": ("mediaconnect", "mediaconnect_backends"),
     "mediapackage": ("mediapackage", "mediapackage_backends"),
+    "mediapackage": ("mediapackage", "mediapackage_backends"),
+    "mediapackage": ("mediapackage", "mediapackage_backends"),
 }
```
Comment (AnyaChamp) [inline moto/backends.py:87]: Just remove this double here (it adds a new one every time you use script/scaffold.py)

**Review by AnyaChamp — CHANGES_REQUESTED** [review]
Lgtm, have you merged in mediapackage, that'll probably get rid of your conflicts.

File: moto/backends.py (hunk 1 above)
Comment (FranckNdame) [reply moto/backends.py:87]: I saw that. i'll remove it once we finish. I think it gets added everytime I run the script. So i'll just clean it up later

**Comment by FranckNdame** [comment]
> Lgtm, have you merged in mediapackage, that'll probably get rid of your conflicts.

No. we need to resolve the conflicts before we merge. **(Currently can't merge)**.
I left it there because I want us to go through it together assuming you haven’t resolved conflicts on the web editor yet and because it involves both our changes.
Github won't let you merge conflicting branches.

**Pull request merged**